kernel/kernel-api/src/main/java/io/delta/kernel/types/StructType.java

```diff
@@ -40,6 +40,11 @@ public StructType() {
   }
 
   public StructType(List<StructField> fields) {
+    validateNoMetadataColumns(
```

Comment:
can you rename this to `validateNoNestedMetadataColumns` and push this logic down? this is otherwise confusing I thought we were validating that there are no metadata columns at all

Reply:
also let's maybe add a second `validateNoDuplicateMetadataColumns` validation? `addMetadataColumn` doesn't stop you from adding duplicates via one of the other add methods

Reply:
and add tests for these in `MetadataColumnSuite` (I think you're already testing most of this yay!)

Reply:
I changed the validation logic because @scovich specifically argued to redesign it this way. See https://github.com/delta-io/delta/pull/5027#discussion_r2308110446. Please discuss with them if you have a strong opinion here.

Regarding the no duplicate check: I also added a check to the regular `add` method that throws an error if you try to add a metadata column and that already exists.

Reply:
For the duplicate check - can we still just do so on the constructor? We only need to cover top level fields and this ensures we enforce this for all struct types, and reduces duplication across the various add methods.